instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/telemetry/PulsarBatchRequest.java

```diff
@@ -30,7 +30,10 @@ private static String getTopicName(Messages<?> messages) {
       if (topicName == null) {
         topicName = name;
       } else if (!topicName.equals(name)) {
-        return null;
+        // this is a partitioned topic
+        // persistent://public/default/test-partition-0 persistent://public/default/test-partition-1
+        // return persistent://public/default/test
+        return topicName.substring(0, topicName.lastIndexOf("-partition-"));
```

Comment:
I'm a little surprise to see this as part of this change. Does it somehow prevent the metrics from getting the right attribute or something?

Reply:
When using a PartitionTopic with the partition count greater than 1, this logic will trigger. I have submitted  a unit test to cover this scenario.

Reply:
```java
TopicName.get(topicName).getPartitionedTopicName()
```
I replaced it with this, which will perform satety side check.